plugins/node/instrumentation-fs/test/fsPromises.test.ts

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { context, trace, SpanStatusCode, SpanKind } from '@opentelemetry/api';
+import { AsyncHooksContextManager } from '@opentelemetry/context-async-hooks';
+import {
+  BasicTracerProvider,
+  InMemorySpanExporter,
+  ReadableSpan,
+  SimpleSpanProcessor,
+} from '@opentelemetry/sdk-trace-base';
+import * as assert from 'assert';
+import Instrumentation from '../src';
+import * as sinon from 'sinon';
+import type * as FSPromisesType from 'fs/promises';
+import tests, { FsFunction, TestCase, TestCreator } from './definitions';
+import type { FPMember, EndHook } from '../src/types';
+
+const supportsPromises =
+  parseInt(process.versions.node.split('.')[0], 10) >= 14;
+
+const TEST_ATTRIBUTE = 'test.attr';
+const TEST_VALUE = 'test.attr.value';
+
+const endHook = <EndHook>sinon.spy((fnName, { args, span }) => {
+  span.setAttribute(TEST_ATTRIBUTE, TEST_VALUE);
+});
+const pluginConfig = {
+  endHook,
+};
+const provider = new BasicTracerProvider();
+const tracer = provider.getTracer('default');
+const memoryExporter = new InMemorySpanExporter();
+provider.addSpanProcessor(new SimpleSpanProcessor(memoryExporter));
+
+if (supportsPromises) {
+  describe('fs/promises instrumentation', () => {
+    let contextManager: AsyncHooksContextManager;
+    let fsPromises: typeof FSPromisesType;
+    let plugin: Instrumentation;
+
+    beforeEach(async () => {
+      contextManager = new AsyncHooksContextManager();
+      context.setGlobalContextManager(contextManager.enable());
+      plugin = new Instrumentation(pluginConfig);
+      plugin.setTracerProvider(provider);
+      plugin.enable();
+      fsPromises = require('fs/promises');
+      assert.strictEqual(memoryExporter.getFinishedSpans().length, 0);
+    });
+
+    afterEach(() => {
+      plugin.disable();
+      memoryExporter.reset();
+      context.disable();
+    });
+
+    const promiseTest: TestCreator = (
+      name: FPMember,
+      args,
+      { error, result, resultAsError = null },
+      spans
+    ) => {
+      const rootSpanName = `${name} test span`;
+      it(`promises.${name} ${error ? 'error' : 'success'}`, async () => {
```

Comment:
Maybe I'm missing something, but isn't this test already exist in the file 'fs.test.ts' ?

Reply:
This is testing `fs/promises`, whereas `fs.test.ts` is testing `fs.promises`.